optuna/study.py

```diff
@@ -95,7 +96,33 @@ def trials(self):
             A list of :class:`~optuna.structs.FrozenTrial` objects.
         """
 
-        return self.storage.get_all_trials(self.study_id)
+        return self._storage.get_all_trials(self.study_id)
+
+    @property
+    def storage(self):
+        # type: () -> storages.BaseStorage
+        """Return the storage object used by the study.
+
+        .. deprecated:: 0.15.0
+            The direct use of storage is deprecated.
+            Please access to storage via study's public APIs
```

Comment:
IMO, `study's public APIs` is a bit ambiguous, and I prefer `study's public methods`.
What do you think?

Reply:
Sounds good.
Fixed: https://github.com/pfnet/optuna/pull/472/commits/d41036adb534cb3dae3586512821973b6ed3a195